report_thesis/src/sections/methodology/testing_validation.tex

```diff
@@ -0,0 +1,223 @@
+In addition, we also compute the metrics for the test set to provide a measure of the model's performance on unseen data.
+Therefore, we have the following metrics for each experiment:
+\begin{enumerate}
+    \item \textbf{Fold-specific \gls{rmse} and Standard Deviation:} For each of the $k$ folds, we calculate both the \gls{rmse} and standard deviation, denoted as \texttt{rmse\_cv\_n} and \texttt{std\_dev\_cv\_n}, where \texttt{n} ranges from 1 to $k$.
+    \item \textbf{Average \gls{rmse} and Standard Deviation:} The overall cross-validation \gls{rmse} (\texttt{rmse\_cv}) and standard deviation (\texttt{std\_dev\_cv}) are computed as the mean of the fold-specific values.
+    \item \textbf{Test Set \gls{rmse} and Standard Deviation:} The \gls{rmse} and standard deviation are also computed for the test set, denoted as \texttt{rmsep} and \texttt{std\_dev}, to provide a measure of the model's performance on unseen data.
+\end{enumerate}
+
+K-fold cross-validation provides a robust estimate of model performance by averaging metrics over multiple folds, reducing variance and offering a clearer picture of the model's generalizability.
+Evaluating the model on a separate test set representative of unseen data ensures that performance metrics accurately reflect the model's generalization capability.
+However, our data partitioning method, which moves the most extreme values into the training data, naturally results in the testing data being closer to the mean of the data distribution, making it easier to predict.
+Therefore, evaluating the model's performance using both the cross-validation metrics (\texttt{rmse\_cv} and \texttt{std\_dev\_cv}) and the test set metrics (\texttt{rmsep} and \texttt{std\_dev}) is crucial.
+The cross-validation metrics provide insights into the model's stability across different subsets, while the test set metrics offer a final measure of performance on truly unseen data, giving a comprehensive assessment of the model's generalizability.
+
+
+\subsubsection{Conclusion}
+In summary, our validation and testing procedures are designed to ensure the robustness and generalizability of our models by employing a combination of k-fold cross-validation and a separate test set.
+The customized k-fold cross-validation algorithm mitigates data leakage by ensuring that data points from the same target are not split across training and testing sets.
+This method, which includes handling extreme values, aligns with established practices and optimally partitions the dataset to maintain a balance between representativeness and model training efficacy.
+By assessing model performance through comprehensive metrics such as \gls{rmse} and standard deviation across both cross-validation folds and a separate test set, we provide a rigorous evaluation framework that underscores the reliability and accuracy of our predictive models.
```

Comment:
```suggestion
By assessing model performance through metrics such as \gls{rmse} and standard deviation across both cross-validation folds and a separate test set, we provide an evaluation framework that underscores the reliability and accuracy of our predictive models.
```

Maybe tone it down a little bit with the adjectives haha

Reply:
😂😂